watertap/unit_models/zero_order/metab_zo.py

```diff
@@ -0,0 +1,111 @@
+###############################################################################
+# WaterTAP Copyright (c) 2021, The Regents of the University of California,
+# through Lawrence Berkeley National Laboratory, Oak Ridge National
+# Laboratory, National Renewable Energy Laboratory, and National Energy
+# Technology Laboratory (subject to receipt of any required approvals from
+# the U.S. Dept. of Energy). All rights reserved.
+#
+# Please see the files COPYRIGHT.md and LICENSE.md for full copyright and license
+# information, respectively. These files are also available online at the URL
+# "https://github.com/watertap-org/watertap/"
+#
+###############################################################################
+"""
+This module contains a zero-order representation of a bioreactor with simple reactions
+(i.e. conversion fractions for key reagents and conversion ratios for other reactive species).
+"""
+
+from pyomo.environ import Constraint, units as pyunits, Var
+from idaes.core import declare_process_block_class
+
+from watertap.core import build_sido_reactive, constant_intensity, ZeroOrderBaseData
+
+# Some more information about this module
+__author__ = "Tim Bartholomew"
+
+
+@declare_process_block_class("MetabZO")
+class MetabZOData(ZeroOrderBaseData):
+    """
+    Zero-Order model for a METAB bioreactor
+    """
+
+    CONFIG = ZeroOrderBaseData.CONFIG()
+
+    def build(self):
+        super().build()
+
+        self._tech_type = "metab"
+        build_sido_reactive(self)
+
+        self._gas_comp = self.config.process_subtype
+
+        # unit variables
+        self.volume = Var(initialize=1,
+                          bounds=(0, None),
+                          units=pyunits.m**3,
+                          doc="Reactor volume")
+        self.hydraulic_retention_time = Var(initialize=1,
+                                            bounds=(0, None),
+                                            units=pyunits.hr,
+                                            doc="Hydraulic residence time")
+        self._fixed_perf_vars.append(self.hydraulic_retention_time)
+
+        @self.Constraint(doc='Constraint for reactor volume based on hydraulic residence time')
+        def eq_reactor_volume(b):
+            return b.volume == (
+                    pyunits.convert(b.get_inlet_flow(0), to_units=pyunits.m ** 3 / pyunits.hour)
+                    * b.hydraulic_retention_time)
+
+        # energy consumption
+        self.electricity = Var(self.flowsheet().time,
+                               initialize=1,
+                               bounds=(0, None),
+                               units=pyunits.kW,
+                               doc="Electricity consumption of unit")
+        self.heat = Var(self.flowsheet().time,
+                        initialize=1,
+                        bounds=(0, None),
+                        units=pyunits.kW,
+                        doc="Thermal energy consumption of unit")
```

Comment:
Technically power consumption, not energy consumption (energy consumption in J or kWh)